src/Raven.Client/Documents/Session/IAttachmentsSessionOperationsAsync.cs

```diff
@@ -30,6 +31,11 @@ public interface IAttachmentsSessionOperationsAsync : IAttachmentsSessionOperati
         /// </summary>
         Task<AttachmentResult> GetAsync(object entity, string name, CancellationToken token = default);
 
+        /// <summary>
+        /// Returns Enumerator of KeyValuePairs of attachment name and stream.
+        /// </summary>
+        Task<IEnumerator<AttachmentEnumeratorResult>> GetAsync(IEnumerable<AttachmentRequest> attachments, CancellationToken token = default);
```

Comment:
Shouldn't we use `IAsyncEnumerator` here like in Streaming?

Reply:
No, we shouldn't.
We are waiting _once_, for the request to complete.
Then we stream the results out.
`IAsyncEnumerator` is for when we are waiting for the next read on the network.
And that is done on the `Stream` itself.